CDATA isn't parsed in Linux?

Looks like the library can't work under Ubuntu 14.04 and Swift 4.0.3? `XMLElement.text` value is empty. Probably related to `XMLParserDelegate`

I prepared a test project: https://github.com/agordeev/SWXMLHash.git

I removed all your tests and added a new one with sample XML I'm using. If you run tests on your Mac, they run just fine. However, when I clone it to my Ubuntu 14.04 server they fail:

```
$ /home/andreygordeev/.swiftenv/versions/4.0.3-RELEASE/usr/bin/swift test
Compile Swift Module 'SWXMLHash' (2 sources)
Compile Swift Module 'SWXMLHashTests' (2 sources)
Compile Swift Module 'SWXMLHashPackageTests' (1 sources)
Linking ./.build/x86_64-unknown-linux/debug/SWXMLHashPackageTests.xctest
Test Suite 'All tests' started at 2018-01-16 06:55:49.559
Test Suite 'debug.xctest' started at 2018-01-16 06:55:49.587
Test Suite 'XMLParsingTests' started at 2018-01-16 06:55:49.587
Test Case 'XMLParsingTests.test1' started at 2018-01-16 06:55:49.588

/home/andreygordeev/SWXMLHash/Tests/SWXMLHashTests/XMLParsingTests.swift:325: error: XMLParsingTests.test1 : XCTAssertTrue failed - 
Test Case 'XMLParsingTests.test1' failed (0.01 seconds)
Test Suite 'XMLParsingTests' failed at 2018-01-16 06:55:49.601
	 Executed 1 test, with 1 failure (0 unexpected) in 0.01 (0.01) seconds
Test Suite 'debug.xctest' failed at 2018-01-16 06:55:49.602
	 Executed 1 test, with 1 failure (0 unexpected) in 0.01 (0.01) seconds
Test Suite 'All tests' failed at 2018-01-16 06:55:49.602
	 Executed 1 test, with 1 failure (0 unexpected) in 0.01 (0.01) seconds
```

As you can see, `print(titleField!.text)` prints empty string on Ubuntu.

--- a/Source/SWXMLHash.swift
+++ b/Source/SWXMLHash.swift
@@ -312,6 +312,18 @@ class LazyXMLParser: NSObject, SimpleXmlParser, XMLParserDelegate {
         current.addText(string)
     }
 
+    func parser(_ parser: XMLParser, foundCDATA CDATABlock: Data) {
+        if !onMatch() {
+            return
+        }
+
+        if let cdataText = String(data: CDATABlock, encoding: String.Encoding.utf8) {
+            let current = parentStack.top()
+
+            current.addText(cdataText)
+        }
+    }
+
     func parser(_ parser: Foundation.XMLParser,
                 didEndElement elementName: String,
                 namespaceURI: String?,
@@ -390,11 +402,11 @@ class FullXMLParser: NSObject, SimpleXmlParser, XMLParserDelegate {
 
         parentStack.drop()
     }
-    
+
     func parser(_ parser: XMLParser, foundCDATA CDATABlock: Data) {
         if let cdataText = String(data: CDATABlock, encoding: String.Encoding.utf8) {
             let current = parentStack.top()
-            
+
             current.addText(cdataText)
         }
     }

--- a/Source/SWXMLHash.swift
+++ b/Source/SWXMLHash.swift
@@ -390,6 +390,14 @@ class FullXMLParser: NSObject, SimpleXmlParser, XMLParserDelegate {
 
         parentStack.drop()
     }
+    
+    func parser(_ parser: XMLParser, foundCDATA CDATABlock: Data) {
+        if let cdataText = String(data: CDATABlock, encoding: String.Encoding.utf8) {
+            let current = parentStack.top()
+            
+            current.addText(cdataText)
+        }
+    }
 }
 
 /// Represents an indexed operation against a lazily parsed `XMLIndexer`

--- a/Tests/SWXMLHashTests/LazyWhiteSpaceParsingTests.swift
+++ b/Tests/SWXMLHashTests/LazyWhiteSpaceParsingTests.swift
@@ -53,11 +53,7 @@ class LazyWhiteSpaceParsingTests: XCTestCase {
     }
 
     func testShouldBeAbleToCorrectlyParseCDATASectionsWithWhitespace() {
-#if os(Linux)
-        print("Skip \(#function) on Linux")
-#else
         XCTAssertEqual(xml!["niotemplate"]["other"].element?.text, "\n        \n  this\n  has\n  white\n  space\n        \n    ")
-#endif
     }
 }
 

--- a/Tests/SWXMLHashTests/WhiteSpaceParsingTests.swift
+++ b/Tests/SWXMLHashTests/WhiteSpaceParsingTests.swift
@@ -52,11 +52,7 @@ class WhiteSpaceParsingTests: XCTestCase {
     }
 
     func testShouldBeAbleToCorrectlyParseCDATASectionsWithWhitespace() {
-#if os(Linux)
-        print("Skip \(#function) on Linux")
-#else
         XCTAssertEqual(xml!["niotemplate"]["other"].element?.text, "\n        \n  this\n  has\n  white\n  space\n        \n    ")
-#endif
     }
 }
 

